java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/server/TaskManagerServer.java

```diff
@@ -16,34 +16,66 @@
 
 package com._4paradigm.openmldb.taskmanager.server;
 
-import com._4paradigm.openmldb.proto.TaskManager;
-import com.baidu.brpc.protocol.BrpcMeta;
+import com._4paradigm.openmldb.taskmanager.zk.FailoverWatcher;
+import lombok.extern.slf4j.Slf4j;
+import com._4paradigm.openmldb.taskmanager.config.TaskManagerConfig;
+import com._4paradigm.openmldb.taskmanager.server.impl.TaskManagerImpl;
+import com.baidu.brpc.server.RpcServer;
+import com.baidu.brpc.server.RpcServerOptions;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
 
-public interface TaskManagerServer {
-    @BrpcMeta(serviceName = "openmldb.taskmanager.TaskManagerServer", methodName = "ShowJobs")
-    TaskManager.ShowJobsResponse ShowJobs(TaskManager.ShowJobsRequest request);
+import java.io.IOException;
 
-    @BrpcMeta(serviceName = "openmldb.taskmanager.TaskManagerServer", methodName = "ShowJob")
-    TaskManager.ShowJobResponse ShowJob(TaskManager.ShowJobRequest request);
+@Slf4j
+public class TaskManagerServer {
+    private static final Log logger = LogFactory.getLog(TaskManagerServer.class);
 
-    @BrpcMeta(serviceName = "openmldb.taskmanager.TaskManagerServer", methodName = "StopJob")
-    TaskManager.StopJobResponse StopJob(TaskManager.StopJobRequest request);
+    public TaskManagerServer() {
+        try {
+            FailoverWatcher failoverWatcher = new FailoverWatcher();
 
-    @BrpcMeta(serviceName = "openmldb.taskmanager.TaskManagerServer", methodName = "DeleteJob")
-    TaskManager.DeleteJobResponse DeleteJob(TaskManager.DeleteJobRequest request);
+            logger.info("The server runs and prepares for leader election");
+            if (failoverWatcher.blockUntilActive()) {
+                logger.info("The server becomes active master and prepare to do business logic");
+                runBrpcServer();
+            }
+            failoverWatcher.close();
+            logger.info("The server exits after running business logic");
 
```

Comment:
If the master lost the lock, it should be serving no more

Reply:
Well, we may handle DISCONNECTED message to check if this is the master and kill itself if disconnected.